maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svnexe/src/main/java/org/apache/maven/scm/provider/svn/svnexe/command/untag/SvnUntagCommand.java

```diff
@@ -0,0 +1,151 @@
+import org.apache.maven.scm.provider.svn.command.SvnCommand;
+import org.apache.maven.scm.provider.svn.repository.SvnScmProviderRepository;
+import org.apache.maven.scm.provider.svn.svnexe.command.SvnCommandLineUtils;
+import org.codehaus.plexus.util.FileUtils;
+import org.codehaus.plexus.util.Os;
+import org.codehaus.plexus.util.cli.CommandLineException;
+import org.codehaus.plexus.util.cli.CommandLineUtils;
+import org.codehaus.plexus.util.cli.Commandline;
+
+/**
+ * SCM-917 provide untag
+ *
+ * scm:untag for provider svn is done by removing the tag dir
+ *
+ * @author Clemens Quoss
+ */
+public class SvnUntagCommand extends AbstractUntagCommand implements SvnCommand
+{
+
+    /** {@inheritDoc} */
+    @Override
+    public ScmResult executeUntagCommand( ScmProviderRepository repo, ScmFileSet fileSet, String tag, String message )
+        throws ScmException
+    {
+        if ( tag == null || tag.trim().isEmpty() )
+        {
+            throw new ScmException( "tag name must be specified" );
+        }
+
+        SvnScmProviderRepository repository = (SvnScmProviderRepository) repo;
+
+        File messageFile = FileUtils.createTempFile( "maven-scm-", ".commit", null );
+
+        try
+        {
+            FileUtils.fileWrite( messageFile.getAbsolutePath(), "UTF-8", message );
+        }
+        catch ( IOException ex )
+        {
+            return new UntagScmResult( null, "Error while making a temporary file for the commit message: "
+                + ex.getMessage(), null, false );
+        }
+
+        Commandline cl = createCommandline( repository, fileSet, tag, messageFile );
+
+        CommandLineUtils.StringStreamConsumer stdout = new CommandLineUtils.StringStreamConsumer();
+
+        CommandLineUtils.StringStreamConsumer stderr = new CommandLineUtils.StringStreamConsumer();
+
+        if ( getLogger().isInfoEnabled() )
+        {
+            getLogger().info( "Executing: " + SvnCommandLineUtils.cryptPassword( cl ) );
+
+            if ( Os.isFamily( Os.FAMILY_WINDOWS ) )
+            {
+                getLogger().info( "Working directory: " + cl.getWorkingDirectory().getAbsolutePath() );
+            }
+        }
+
+        int exitCode;
+
+        try
+        {
+            exitCode = SvnCommandLineUtils.execute( cl, stdout, stderr, getLogger() );
+        }
+        catch ( CommandLineException ex )
+        {
+            throw new ScmException( "Error while executing svn untag command.", ex );
+        }
+        finally
+        {
+            try
+            {
+                FileUtils.forceDelete( messageFile );
+            }
+            catch ( IOException ex )
+            {
+                // ignore
+            }
+        }
+
+        if ( exitCode == 0 )
+        {
+            return new UntagScmResult( cl.toString(), stdout.getOutput(), stderr.getOutput(), true );
+        }
+        else
+        {
+            return new UntagScmResult( cl.toString(), stdout.getOutput(), stderr.getOutput(), false );
```

Comment:
I doubt that something will be written to stdout. Why not use "The svn remove command failed." as does the tag command?

Reply:
Will change this, but shouldn't SvnTagCommand then also state "The svn copy command failed" instead of "The svn tag command failed"?